pallets/messages/src/lib.rs

```diff
@@ -179,16 +191,15 @@ pub mod pallet {
 			schema_id: SchemaId,
 			/// The block number for these messages
 			block_number: T::BlockNumber,
-			/// Number of messages in this block for this schema
-			count: u16,
 		},
 	}
 
 	#[pallet::hooks]
 	impl<T: Config> Hooks<BlockNumberFor<T>> for Pallet<T> {
-		fn on_initialize(current: T::BlockNumber) -> Weight {
-			let prev_block = current - T::BlockNumber::one();
-			Self::move_messages_into_final_storage(prev_block)
+		fn on_initialize(_current: T::BlockNumber) -> Weight {
+			<MessageIndex<T>>::set(0u16);
```

Comment:
reseting temporary to 0

Reply:
What if we do not clear it and keep the index?

Reply:
It will overflow. We just need order in a block to be able to get canonical order. To keep each message size small I would not like to increase existing u16 to `u32` or `u64`

Reply:
Ahh thats right, each message has an index.